src/main/java/org/openscience/cdk/geometry/cip/CIPToolMod.java

```diff
@@ -271,8 +287,45 @@ private static int permParity(final ILigand[] ligands) {
         return (swaps & 0x1) == 0x1 ? -1 : +1;
     
     }
-    
 
-    
-   
+    public static int getSizeOfLargestRingSystem(CdkChemicalImpl chemical) {
+        CdkChemicalImpl copy = chemical.deepCopy();
+        int maxRings = 0;
+        List<Bond> bondsToRemove = new ArrayList<>();
+        for(int i = copy.getBondCount()-1; i >=0; i--) {
+            Bond bond = copy.getBond(i);
+            if(!bond.isInRing() ) {
+                //copy.removeBond(i);
+                bondsToRemove.add(bond);
+            }
+        }
+        bondsToRemove.forEach(b-> copy.removeBond(b));
+
+        if( copy.getBondCount() == 0) {
+            //we have removed all non-ring bonds. When we have no bonds, that means no rings
+            return 0;
+        }
+
+        for(int i = copy.getAtomCount()-1; i >=0; i--) {
+            Atom atom = copy.getAtom(i);
+            if(atom.getBondCount() == 0) {
+                //System.out.printf("atom %d of symbol %s has no bonds and will be deleted\n", i, atom.getSymbol());
+                copy.removeAtom(i);
```

Comment:
Let's do same trick here to add to a list and tehm remove them all. I think otherwise there are some other caches that can go wrong.